survival_guide/3_operations.adoc

```diff
@@ -590,32 +590,45 @@ Our recommended configuration for using the Spark History Server with Fusion is
 [source,yaml]
 ----
 gcs:
-    enableGCS: false
-    secret: history-secrets
-    key: [SECRET_KEY_NAME].json
-    logDirectory: gs://[BUCKET_NAME]
+  enableGCS: true
+  secret: history-secrets
+  key: sparkhistory.json
+  logDirectory: gs://[BUCKET_NAME]
 service:
-    type: ClusterIP
-    port: 18080
+  type: ClusterIP
+  port: 18080
 ----
-We override the default `service.type` of `LoadBalancer` with `ClusterIP` to keep the History Server from being accessible to outside connections without port-forwarding.
+The `service` key sets up the history server on an internal IP within your cluster but does not create an external LoadBalancer (which is the default setting in the Spark History Server Helm chart). This prevents the server from being exposed to outside access by default. We’ll look at how to access the history server shortly.
+
+The `key` and `secret` fields provide the Spark History Server with the details of where it will find an account with access to the Google Cloud Storage bucket given in `logDirectory`. In the following example, we're going to set up a new service account that will be shared between the Spark History Server and the Spark driver/executors for both viewing and writing logs.
 
-You may need to set up your secret for full access to the cloud bucket:
 [source,bash]
 ----
-$ export ACCOUNT_NAME=[SECRET_KEY_NAME]
+$ export ACCOUNT_NAME=sparkhistory
 $ export GCP_PROJECT_ID=[PROJECT_ID]
 $ gcloud iam service-accounts create ${ACCOUNT_NAME} --display-name "${ACCOUNT_NAME}"
 $ gcloud iam service-accounts keys create "${ACCOUNT_NAME}.json" --iam-account "${ACCOUNT_NAME}@${GCP_PROJECT_ID}.iam.gserviceaccount.com"
 $ gcloud projects add-iam-policy-binding ${GCP_PROJECT_ID} --member "serviceAccount:${ACCOUNT_NAME}@${GCP_PROJECT_ID}.iam.gserviceaccount.com" --role roles/storage.admin
 $ gsutil iam ch serviceAccount:${ACCOUNT_NAME}@${GCP_PROJECT_ID}.iam.gserviceaccount.com:objectAdmin gs://[BUCKET_NAME]
 ----
-The service key sets up the history server on an internal IP within your cluster but does not create a LoadBalancer (which is the default setting in the Spark History Server Helm chart). This prevents the server from being exposed to outside access by default. We’ll look at how to access the history server shortly.
+
+First, we use `create` to create a new service account, and then `keys create` to create a JSON keypair which we will shortly upload into our cluster as a Kubernetes secret. We then give our service account the `storage/admin` role, allowing it to perform create and view operations, and the final `gsutil` command applies our service account to our chosen bucket. If you have an existing service account you wish to use instead, you can skip the `create` command, though you will still need to create the JSON keypair and ensure that the existing account can read and write to the log bucket.
```

Comment:
When you give command line things to run, the explanation for what it's going to do should be before each command, not after. Basically "here's what we're gonna do next" and then "here's us doing it", rather than "here's us doing something you don't understand" and then "here's what we just did". 

It can be broken up into smaller paragraphs (explanation -> command, explanation -> command) to help with readability also.